infrastructure-components.yaml

```diff
@@ -0,0 +1,833 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cluster-api-provider-outscale-proxy-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-api-provider-outscale-proxy-role
+subjects:
+- kind: ServiceAccount
+  name: cluster-api-provider-outscale-controller-manager
+  namespace: cluster-api-provider-outscale-system
+---
+apiVersion: v1
+data:
+  controller_manager_config.yaml: |
+    apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
+    kind: ControllerManagerConfig
+    health:
+      healthProbeBindAddress: :8081
+    metrics:
+      bindAddress: 127.0.0.1:8080
+    webhook:
+      port: 9443
+    leaderElection:
+      leaderElect: true
+      resourceName: 4b982d66.cluster.x-k8s.io
+kind: ConfigMap
+metadata:
+  name: cluster-api-provider-outscale-manager-config
+  namespace: cluster-api-provider-outscale-system
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: cluster-api-provider-outscale-controller-manager-service
+  namespace: cluster-api-provider-outscale-system
+spec:
+  ports:
+  - name: https
+    port: 8443
+    protocol: TCP
+    targetPort: https
+  selector:
+    control-plane: controller-manager
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: cluster-api-provider-outscale-controller-manager
+  namespace: cluster-api-provider-outscale-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      control-plane: controller-manager
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+      labels:
+        control-plane: controller-manager
+    spec:
+      containers:
+      - args:
+        - --secure-listen-address=0.0.0.0:8443
+        - --upstream=http://127.0.0.1:8080/
+        - --logtostderr=true
+        - --v=0
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
+        name: kube-rbac-proxy
+        ports:
+        - containerPort: 8443
+          name: https
+          protocol: TCP
+        resources:
+          limits:
+            cpu: 500m
+            memory: 128Mi
+          requests:
+            cpu: 5m
+            memory: 64Mi
+      - args:
+        - --health-probe-bind-address=:8081
+        - --metrics-bind-address=127.0.0.1:8080
+        - --leader-elect
+        command:
+        - /manager
+        env:
+        - name: OSC_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              key: access_key
+              name: cluster-api-provider-outscale
+              optional: true
+        - name: OSC_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              key: secret_key
+              name: cluster-api-provider-outscale
+              optional: true
+        image: 042b4721a38342028d65c28be2b30e64-157001637.eu-west-2.lbu.outscale.com:5000/controller:latest
```

Comment:
should be dynamic, no ?

Reply:
Ok but could you but a real URL here ? Like the future URL in the docker registry ?

Reply:
Ok done
values in dockerhub